app/javascript/packages/url/index.ts

```diff
@@ -2,25 +2,15 @@
  * Given a URL or a string fragment of search parameters and an object of parameters, returns a
  * new URL or search parameters with the parameters added.
  *
- * @param urlOrParams Original URL or search parameters.
+ * @param url Original URL, path, or search parameters.
  * @param params Search parameters to add.
  *
  * @return Modified URL or search parameters.
  */
-export function addSearchParams(urlOrParams: string, params: Record<string, any>): string {
-  let parsedURLOrParams: URL | URLSearchParams;
-  let searchParams: URLSearchParams;
-
-  try {
-    parsedURLOrParams = new URL(urlOrParams);
```

Comment:
is there a reason we moved away from the `URL` class? I'd love to be able to rely on something like that instead of having to build our own

Reply:
> is there a reason we moved away from the `URL` class? I'd love to be able to rely on something like that instead of having to build our own

I went through a few iterations of this. The main problem with the URL class is that this function is meant to be quite flexible with its input values (absolute URLs, paths, querystring-only), and it's hard to map each of those to a corresponding return value of the same form using URL.

That being said, I couldn't actually find any existing usage of the querystring parameter input, so we could adjust this to assume that both the input and output would be a full URL.

Reply:
Also, to be more specific about why it was changed here: We had been [initializing `startOverURL` and `cancelURL` with paths](https://github.com/18F/identity-idp/blob/0e8c7e136f6552d17f13a77700868d84b133c709/app/views/idv/shared/_document_capture.html.erb#L34-L35). A `URL` constructor won't parse a path fragment on its own (`new URL('/foo')` throws). We could adjust this to parse relative to the current URL (e.g. `new URL('/foo', window.location.href)`), but at that point is when it becomes harder to determine how to return the expected value (`addSearchParams('/foo', { a: 1 }) === '???'`).

Reply:
> That being said, I couldn't actually find any existing usage of the querystring parameter input, so we could adjust this to assume that both the input and output would be a full URL.

Simplifying supports simplifies the implementation! 0f3f5112e5b72bb56691157f809b0e695cb4b3c9